src/com/dabomstew/pkrandom/newgui/BatchRandomizationSettingsDialog.java

```diff
@@ -0,0 +1,145 @@
+import java.awt.event.*;
+import java.io.File;
+
+public class BatchRandomizationSettingsDialog extends JDialog {
+    private JPanel mainPanel;
+    private JButton okButton;
+    private JButton cancelButton;
+    private JCheckBox enableBatchRandomizationCheckBox;
+    private JSpinner numberOfRandomizedROMsSpinner;
+    private JSpinner startingIndexSpinner;
+    private JTextField fileNamePrefixTextField;
+    private JCheckBox generateLogFilesCheckBox;
+    private JCheckBox autoAdvanceIndexCheckBox;
+    private JButton chooseDirectoryButton;
+    private JLabel outputDirectoryLabel;
+
+    private JFileChooser outputDirectoryFileChooser;
+
+    private final BatchRandomizationSettings currentSettings;
+
+    public BatchRandomizationSettings getCurrentSettings() {
+        return this.currentSettings;
+    }
+
+    public BatchRandomizationSettingsDialog(JFrame parent, BatchRandomizationSettings currentSettings) {
+        super(parent, true);
+        add(mainPanel);
+        java.util.ResourceBundle bundle = java.util.ResourceBundle.getBundle("com/dabomstew/pkrandom/newgui/Bundle");
+        setTitle(bundle.getString("BatchRandomizationSettingsDialog.title"));
+        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
+
+        getRootPane().setDefaultButton(okButton);
+
+        this.currentSettings = currentSettings.clone();
+
+        initializeControls();
+        setLocationRelativeTo(parent);
+        pack();
+        setVisible(true);
+    }
+
+    private void onOK() {
+        updateSettings();
+        setVisible(false);
+    }
+
+    private void onCancel() {
+        // add your code here if necessary
+        setVisible(false);
+    }
+
+    private void initializeControls() {
+        outputDirectoryFileChooser = new JFileChooser();
+        okButton.addActionListener(e -> onOK());
+        cancelButton.addActionListener(e -> onCancel());
+
+        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
+        addWindowListener(new WindowAdapter() {
+            public void windowClosing(WindowEvent e) {
+                onCancel();
+            }
+        });
+        mainPanel.registerKeyboardAction(e -> onCancel(), KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
+                JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
+
+        SpinnerNumberModel numberOfRandomizedROMsModel = new SpinnerNumberModel(1,1, Integer.MAX_VALUE, 1);
+        numberOfRandomizedROMsSpinner.setModel(numberOfRandomizedROMsModel);
+
+        SpinnerNumberModel startingIndexModel = new SpinnerNumberModel(1,0, Integer.MAX_VALUE, 1);
+        startingIndexSpinner.setModel(startingIndexModel);
+
+        chooseDirectoryButton.addActionListener(e -> {
+            int selectionResult = outputDirectoryFileChooser.showDialog(this, "Select");
+            if (selectionResult == JFileChooser.APPROVE_OPTION) {
+                outputDirectoryFileChooser.setCurrentDirectory(new File(currentSettings.getOutputDirectory()));
```

Comment:
```suggestion
                outputDirectoryFileChooser.setCurrentDirectory(outputDirectoryFileChooser.getSelectedFile());
```
This makes it so that if you select a directory in the file chooser and then open the file chooser again without pressing the "OK" button, it'll go to the directory you selected instead of the directory that's currently stored in the settings.